src/common.js

```diff
@@ -151,13 +151,13 @@ export function resolveUrl (relUrl, parentUrl) {
   return resolveIfNotPlainOrUrl(relUrl, parentUrl) || (relUrl.indexOf(':') !== -1 ? relUrl : resolveIfNotPlainOrUrl('./' + relUrl, parentUrl));
 }
 
-function resolveAndComposePackages (packages, outPackages, baseUrl, parentMap) {
+function resolveAndComposePackages (packages, outPackages, baseUrl, parentMap, parentUrl) {
   for (let p in packages) {
-    const resolvedLhs = resolveIfNotPlainOrUrl(p, baseUrl) || p;
+    const resolvedLhs = resolveIfNotPlainOrUrl(p, parentUrl != undefined ? parentUrl : baseUrl) || p;
     let target = packages[p];
     if (typeof target !== 'string') 
       continue;
-    const mapped = resolveImportMap(parentMap, resolveIfNotPlainOrUrl(target, baseUrl) || target, baseUrl);
+    const mapped = resolveImportMap(parentMap, resolveIfNotPlainOrUrl(target, baseUrl) || target, parentUrl);
```

Comment:
The target itself must always be resolved relative to the `baseUrl` not the scope Url.